### PR TITLE
added controllers for registration, delete user, display all users, a…

### PR DESCRIPTION
**Pull request by Nick97-git**
…dd of product, display all products, add product to shopping cart, display all products in shopping cart of user

### CODE REVIEW

File: src/main/java/mate/academy/internet/shop/controllers/AddProductController.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internet.shop.lib.Injector;
+import mate.academy.internet.shop.model.Product;
+import mate.academy.internet.shop.service.ProductService;
+
+public class AddProductController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate.academy.internet.shop");
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/controllers/AddProductController.java]:
```suggestion
    private static final Injector INJECTOR = Injector.getInstance("mate.academy.internet.shop");
```

File: src/main/java/mate/academy/internet/shop/controllers/AddProductToCartController.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internet.shop.lib.Injector;
+import mate.academy.internet.shop.model.ShoppingCart;
+import mate.academy.internet.shop.service.ProductService;
+import mate.academy.internet.shop.service.ShoppingCartService;
+
+public class AddProductToCartController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate.academy.internet.shop");
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/controllers/AddProductToCartController.java]:
```suggestion
    private static final Injector INJECTOR = Injector.getInstance("mate.academy.internet.shop");
```

File: src/main/java/mate/academy/internet/shop/controllers/DeleteUserController.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internet.shop.lib.Injector;
+import mate.academy.internet.shop.service.UserService;
+
+public class DeleteUserController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate.academy.internet.shop");
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/controllers/DeleteUserController.java]:
```suggestion
    private static final Injector INJECTOR = Injector.getInstance("mate.academy.internet.shop");
```

File: src/main/java/mate/academy/internet/shop/controllers/InjectDataController.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internet.shop.lib.Injector;
+import mate.academy.internet.shop.model.Product;
+import mate.academy.internet.shop.model.User;
+import mate.academy.internet.shop.service.ProductService;
+import mate.academy.internet.shop.service.UserService;
+
+public class InjectDataController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate.academy.internet.shop");
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/controllers/InjectDataController.java]:
```suggestion
    private static final Injector INJECTOR = Injector.getInstance("mate.academy.internet.shop");
```

File: src/main/java/mate/academy/internet/shop/controllers/ProductsController.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internet.shop.lib.Injector;
+import mate.academy.internet.shop.model.Product;
+import mate.academy.internet.shop.service.ProductService;
+
+public class ProductsController extends HttpServlet {
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/controllers/ProductsController.java]:
```suggestion
public class GetAllProductsController extends HttpServlet {
```

File: src/main/java/mate/academy/internet/shop/controllers/ProductsController.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    private static final Injector injector = Injector.getInstance("mate.academy.internet.shop");
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/controllers/ProductsController.java]:
```suggestion
    private static final Injector INJECTOR = Injector.getInstance("mate.academy.internet.shop");
```

File: src/main/java/mate/academy/internet/shop/controllers/RegistrationController.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internet.shop.lib.Injector;
+import mate.academy.internet.shop.model.User;
+import mate.academy.internet.shop.service.UserService;
+
+public class RegistrationController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate.academy.internet.shop");
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/controllers/RegistrationController.java]:
```suggestion
    private static final Injector INJECTOR = Injector.getInstance("mate.academy.internet.shop");
```

File: src/main/java/mate/academy/internet/shop/controllers/ShoppingCartController.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internet.shop.lib.Injector;
+import mate.academy.internet.shop.model.ShoppingCart;
+import mate.academy.internet.shop.service.ShoppingCartService;
+
+public class ShoppingCartController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("mate.academy.internet.shop");
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/controllers/ShoppingCartController.java]:
```suggestion
    private static final Injector INJECTOR = Injector.getInstance("mate.academy.internet.shop");
```

File: src/main/java/mate/academy/internet/shop/controllers/UsersController.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internet.shop.lib.Injector;
+import mate.academy.internet.shop.model.User;
+import mate.academy.internet.shop.service.UserService;
+
+public class UsersController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate.academy.internet.shop");
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/controllers/UsersController.java]:
```suggestion
    private static final Injector INJECTOR = Injector.getInstance("mate.academy.internet.shop");
```

Comment (Nick97-git) [reply src/main/java/mate/academy/internet/shop/controllers/UsersController.java]: done

File: src/main/java/mate/academy/internet/shop/controllers/ShoppingCartController.java (hunk 8 above)
Comment (Nick97-git) [reply src/main/java/mate/academy/internet/shop/controllers/ShoppingCartController.java]: done

File: src/main/java/mate/academy/internet/shop/controllers/RegistrationController.java (hunk 7 above)
Comment (Nick97-git) [reply src/main/java/mate/academy/internet/shop/controllers/RegistrationController.java]: done

File: src/main/java/mate/academy/internet/shop/controllers/ProductsController.java (hunk 6 above)
Comment (Nick97-git) [reply src/main/java/mate/academy/internet/shop/controllers/ProductsController.java]: done

File: src/main/java/mate/academy/internet/shop/controllers/ProductsController.java (hunk 5 above)
Comment (Nick97-git) [reply src/main/java/mate/academy/internet/shop/controllers/ProductsController.java]: done

File: src/main/java/mate/academy/internet/shop/controllers/InjectDataController.java (hunk 4 above)
Comment (Nick97-git) [reply src/main/java/mate/academy/internet/shop/controllers/InjectDataController.java]: done

File: src/main/java/mate/academy/internet/shop/controllers/DeleteUserController.java (hunk 3 above)
Comment (Nick97-git) [reply src/main/java/mate/academy/internet/shop/controllers/DeleteUserController.java]: done

File: src/main/java/mate/academy/internet/shop/controllers/AddProductToCartController.java (hunk 2 above)
Comment (Nick97-git) [reply src/main/java/mate/academy/internet/shop/controllers/AddProductToCartController.java]: done

File: src/main/java/mate/academy/internet/shop/controllers/AddProductController.java (hunk 1 above)
Comment (Nick97-git) [reply src/main/java/mate/academy/internet/shop/controllers/AddProductController.java]: done

**Review by Tarabarnik — APPROVED** [review]

**Pull request merged**